app/scripts/modules/core/src/slack/SlackChannelSelector.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import * as Select from 'react-select';
+import VirtualizedSelect from 'react-virtualized-select';
+
+import { ISlackChannel, SlackReader } from './SlackReader';
+
+export interface ISlackChannelSelectorProps {
+  channel: ISlackChannel;
+  callback: (name: string, value: any) => void;
+}
+
+export interface ISlackChannelSelectorState {
+  channels: ISlackChannel[];
+  selected: ISlackChannel;
+  loading: boolean;
+}
+
+export default class SlackChannelSelector extends React.Component<
```

Comment:
i'm curious why you chose to use a class component over a hooks component??

Reply:
I think my main hesitation was ensuring that the API was only called one time, rather than after each render. I can refactor to a functional component with `useEffect` & a second parameter.

Reply:
If you refactor to functional component, use the `useData` or `useLatestPromise` hook!